File: caesar.rb
#Caesar Cypher - using no %, and no orm.
#Basically all you need is to rotate the array at the offset

def caesar(string, offset)
    arr = string.downcase.split('')
    alphabet = ("a".."z").to_a
    encoded = alphabet.rotate(offset)
    library = {" ": " ", ".": ".", "!": "!", ",": ",", "-": "-"}
    alphabet.each_with_index do |letter, i| 
        library[letter] = encoded[i]
    end
    result = arr.map{|el| library[el] }
    result.join('')
end


puts "First cypher - with an offset of 3, wxyz should be zabc"
puts "Result: #{caesar("wxyz", 3)}"
puts "Second cypher - with an offset of 4, ATTACKATONCE should be EXXEGOEXSRGI"
puts "Result: #{caesar("ATTACKATONCE", 4)}"

File: caesar.js
/**
 * Caesar Cipher for JavaScript, using a custom rotation subroutine.
 */

import {ALPHABET} from './constants';

const rotation = (offset) => {
    const left = ALPHABET.slice(offset);
    const right = ALPHABET.slice(0, offset);
    return left.concat(right);
}

const caesar = (string, offset) => {
    array = string.split('')
    const library = {' ': ' ', '.':'.', ',': ',', '!': '!', '-': '-'};
    const encoded = rotation(offset);
    for (let i =0; i < ALPHABET.length; i++) {
        library[ALPHABET[i]] = encoded[i];
    }
    const result = array.map( (letter) => {
        return library[letter]
    })
    return result.join('');
}

console.log("First cypher - with an offset of 3, wxyz should be zabc");
console.log(`Result: ${caesar("wxyz", 3)}`)